docs/integrations/sources/adroll.md

```diff
@@ -0,0 +1,47 @@
+# AdRoll
+This is the connector for AdRoll
+
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `Authorization` | `string` | Person Access Token (Authorization).  |  |
+| `client_id` | `string` | Client ID.  |  |
+| `client_secret` | `string` | Client secret.  |  |
+| `client_refresh_token` | `string` | Refresh token.  |  |
+| `oauth_access_token` | `string` | Access token. The current access token. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+| `oauth_token_expiry_date` | `string` | Token expiry date. The date the current access token expires in. This field might be overridden by the connector based on the token refresh endpoint response. |  |
+| `advertisable` | `string` | Advertisable EID .  |  |
+| `adgroup_eid` | `string` | AdGroup EID.  |  |
+| `campaign_eid` | `string` | Campaign EID.  |  |
```

Comment:
This doesn't look right — we should fetch all campaigns, strategies, asgroups, etc.

Reply:
> This doesn't look right — we should fetch all campaigns, strategies, asgroups, etc.

I think you missed my comment on the issue ticket 😅 
 I was getting errors for these streams so they were not added 
1) To fetch all ads belonging to an advertisable  : [Ads](https://developers.adroll.com/docs/adroll-ad-api/1/routes/ads/get)   ---->INVALID SCHEMA 
2) To fetch an Ad by EID [Ad by EId](https://developers.adroll.com/docs/adroll-ad-api/1/routes/ads/%7Beid%7D/get).  ---->USER MUST BE ADMIN 
3) Fetch Assets belonging to the Advertisable [Assets](https://developers.adroll.com/docs/adroll-ad-api/1/routes/assets/get).  ---->INVALID SCHEMA 
4) Fetch segment information by EID [Segment by EId](https://developers.adroll.com/docs/adroll-audience-api/1/routes/segments/%7Bsegment_eid%7D/get) ---->Error 500 something went wrong 


P.S. I tried to get around with these , but I was not successful

I've added the stream to fetch all strategies since a direct endpoint was available , but we don't have specific endpoints to fetch all campaigns and adgroups etc ...
For doing this we will have to use the Reporting API 
Will need to add specific query in the request bodies

Should I proceed ahead with this ?

Reply:
That's exactly the tricky part about making connectors. If you don't have solid access to an account with enough data and high enough level of permissions to make and validate a connector that makes sense, you should not attempt to half-ass a connector. 

If a connector requires configuration on the level that would mean that someone must manually add a connection for each strategy, that's crazy, and borderline not useful.